### PR TITLE
Add tests for HttpListenerResponse header behaviour

### PR DESCRIPTION
**Pull request by hughbe**
Contributes to #13618

### CODE REVIEW

File: src/System.Net.HttpListener/tests/HttpListenerResponseTests.Headers.cs [hunk 1]
```diff
@@ -791,13 +809,39 @@ public static IEnumerable<object[]> ProtocolVersion_Set_TestData()
                 };
                 response.Headers = headers;
                 Assert.Equal(headers, response.Headers);
+                //[ActiveIssue(20425, TargetFrameworkMonikers.Netcoreapp)]
```
Comment (stephentoub) [inline src/System.Net.HttpListener/tests/HttpListenerResponseTests.Headers.cs]: What is this comment referring to?

File: src/System.Net.HttpListener/tests/HttpListenerResponseTests.Headers.cs [hunk 2]
```diff
@@ -808,5 +852,29 @@ public static IEnumerable<object[]> ProtocolVersion_Set_TestData()
...
+            HttpListenerResponse response = await GetResponse();
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("value", () => response.Headers["name"] = s_longString);
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("value", () => response.Headers.Set("name", s_longString));
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("value", () => response.Headers.Add("name", s_longString));
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("value", () => response.Headers.Add($"name:{s_longString}"));
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("value", () => response.Headers.Add(HttpResponseHeader.Age, s_longString));
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
+        [ActiveIssue(20425, TargetFrameworkMonikers.Netcoreapp)]
```
Comment (stephentoub) [inline src/System.Net.HttpListener/tests/HttpListenerResponseTests.Headers.cs]: This issue is about long header values.  Is it the right issue?  Aren't these really just another manifestation of the existing issue https://github.com/dotnet/corefx/issues/18784?

**Comment by danmosemsft** [comment]
@dotnet-bot test Innerloop OSX10.12 Release x64 Build and Test  please (no log)

**Comment by danmosemsft** [comment]
@stephentoub any reason we didn't merge this yet

**Comment by stephentoub** [comment]
> any reason we didn't merge this yet

I was waiting for Hugh to address my feedback... didn't notice he's done so.

**Comment by hughbe** [comment]
Sorry. I thought GitHub notifies you when I push a commit

**Comment by stephentoub** [comment]
> Sorry. I thought GitHub notifies you when I push a commit

No worries.  It normally does.  It probably did and I just missed it.

**Pull request merged**